Test/Pester.Tests.ps1

```diff
@@ -0,0 +1,429 @@
+    [Parameter(Mandatory,Position=0)]
+    [String]`$One,
+
+    [Parameter(Mandatory,Position=1)]
+    [String]`$Two,
+
+    [Parameter(Mandatory)]
+    [String]`$Three,
+
+    [Parameter(Mandatory)]
+    [String]`$Four
+)
+
+Describe 'ArgTest' {
+    It 'should pass them' {
+        `$One | Should -Be '$($oneValue)'
+        `$Two | Should -Be '$($twoValue)'
+        `$Three | Should -Be '$($threeValue)'
+        `$Four | Should -Be '$($fourValue)'
+    }
+}
+"@
+        GivenTestFile 'ArgTest.ps1' $testContent
+        GivenTestFile 'Arg2Test.ps1' $testContent
+        $taskParameter['Container'] = @{
+            Path = ('ArgTest.ps1', 'Arg2Test.ps1');
+            Data = @{
+                One = $oneValue;
+                Two = $twoValue;
+                Three = $threeValue;
+                Four = $fourValue;
+            }
+        }
+        WhenPesterTaskIsInvoked -AsJob
+        ThenDidNotFail
+    }
+}
+
+Describe 'Pester5.when passing an array list'{
+    AfterEach { Reset }
+    It 'should get converted correctly'{
+        Init
+        GivenTestFile 'PassingTests.ps1' @'
+Describe 'Failing' {
+    It 'should fail' {
+        $false | Should -BeFalse
+    }
+}
+'@
+        GivenTestFile 'PassingTests2.ps1' @'
+Describe 'Passing' {
+    It 'should pass' {
+        $true | Should -BeTrue
+    }
+}
+'@
+
+        $pathArrayList = [System.Collections.ArrayList]::new()
+        $pathArrayList.Add('PassingTests.ps1')
+        $pathArrayList.Add('PassingTests2.ps1')
+
+        WhenPesterTaskIsInvoked -AsJob -WithArgument @{
+            Run = @{
+                Path = $pathArrayList;
+            };
+        }
+        ThenDidNotFail
+
+    }
+}
+
+Describe 'Pester5.when passing a string boolean value'{
+    AfterEach { Reset }
+    It 'should get converted correctly and return test results'{
+        Init
+        GivenTestFile 'PassingTests.ps1' @'
+Describe 'Passing' {
+    It 'should pass' {
+        $true | Should -BeTrue
+    }
+}
+'@
+        WhenPesterTaskIsInvoked -AsJob -WithArgument @{
+            Run = @{
+                Path = 'PassingTests.ps1';
+            };
+            TestResult = @{
+                Enabled = 'True';
+                OutputPath = GetOutputPath;
+                OutputFormat = 'NUnitXml';
+            };
+        }
+        ThenDidNotFail -AndPublishedTestResult
+    }
+}
+
+Describe 'Pester5.when passing a script block'{
+    AfterEach{ Reset }
+    It 'should pass script block correctly'{
+        Init
+        $oneValue = [Guid]::NewGuid()
+        $scriptBlock = {
+            param(
+                [Parameter(Mandatory)]
+                [String] $One
+            )
+            Describe 'Passing' {
+                It 'should pass' {
+                    $One | Should -Be $oneValue
+                }
+            }
+        }
+        $taskParameter['Container'] = @{
+            ScriptBlock =  $scriptBlock;
+            Data = @{
+                One = $oneValue
+            }
+        }

```

Comment:
We should add some tests where we define the `Pester` task via a `whiskey.yml` with the `Container` property specified as a YAML dictionary so we can test that the dictionary/hashtable comes through correctly to `New-PesterContainer` when being read from the YAML.